### DISCUSSION TITLE
Different `#[AsCronTask]` per environment

### DISCUSSION THREAD

**Discussion by VincentLanglet** [symfony/symfony#54231] [Q&A] [answered]
Is there a way to pass the environment when using `#[AsCronTask]` attribute ?

For instance:
- If I want a cron to run every day in production, but every Monday on staging ; how do I do ?
- If I want a cron to run every day in staging, but not in production ; how do I do ?

I would have expect something like
```
#[AsCronTask('*', env: 'prod')]
```
or
```
#[When('prod', #[AsCronTask('*')])]
```

But I didn't find any correct syntax so far.

**Comment by MatTheCat** [comment, marked as answer]
AFAIK this is not possible using `AsCronTask`; you would need to change your service’s `scheduler.task` tag `frequency` (or `expression`) attribute in the configuration or a compiler pass.
